storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4215,7 +4474,48 @@ public boolean isTopologyNameAllowed(String name) throws AuthorizationException,
             throw new RuntimeException(e);
         }
     }
-    
+
+    @Override
+    public SupervisorAssignments getSupervisorAssignments(String node) throws AuthorizationException, TException {
+        try {
```

Comment:
Would be good to have some authorization checks here.  Not super critical because the data is public, but would be good to tie it into out existing authentication system.